ygot/struct_validation_map_test.go

```diff
@@ -872,3 +872,385 @@ func TestMergeStructJSON(t *testing.T) {
+	}, {
+		name: "non-pointer",
+		inA:  reflect.ValueOf("4-pines-indian-summer-ale"),
+		inB:  reflect.ValueOf("james-squire-150-lashes"),
+	}}
+
+	for _, tt := range tests {
+		if err := copyStruct(tt.inA, tt.inB); err == nil {
+			t.Errorf("%s: copyStruct(%v, %v): did not get nil error, got: %v, want: nil", tt.name, tt.inA, tt.inB, err)
+		}
+	}
+}
+
+func TestCopyStruct(t *testing.T) {
+	tests := []struct {
+		name    string
+		inSrc   GoStruct
+		inDst   GoStruct
+		wantDst GoStruct
+		wantErr bool
+	}{{
+		name:    "simple string pointer",
+		inSrc:   &copyTest{StringField: String("anchor-steam")},
+		inDst:   &copyTest{},
+		wantDst: &copyTest{StringField: String("anchor-steam")},
+	}, {
+		name: "uint and string pointer",
+		inSrc: &copyTest{
+			StringField: String("fourpure-juicebox"),
+			Uint32Field: Uint32(42),
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			StringField: String("fourpure-juicebox"),
+			Uint32Field: Uint32(42),
+		},
+	}, {
+		name: "struct pointer with single field",
+		inSrc: &copyTest{
+			StringField: String("lagunitas-aunt-sally"),
+			StructPointer: &copyTest{
+				StringField: String("deschutes-pinedrops"),
+			},
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			StringField: String("lagunitas-aunt-sally"),
+			StructPointer: &copyTest{
+				StringField: String("deschutes-pinedrops"),
+			},
+		},
+	}, {
+		name: "struct pointer with multiple fields",
+		inSrc: &copyTest{
+			StringField: String("allagash-brett"),
+			Uint32Field: Uint32(84),
+			StructPointer: &copyTest{
+				StringField: String("brooklyn-summer-ale"),
+				Uint32Field: Uint32(128),
+			},
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			StringField: String("allagash-brett"),
+			Uint32Field: Uint32(84),
+			StructPointer: &copyTest{
+				StringField: String("brooklyn-summer-ale"),
+				Uint32Field: Uint32(128),
+			},
+		},
+	}, {
+		name: "enum value",
+		inSrc: &copyTest{
+			EnumValue: EnumTypeValue,
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			EnumValue: EnumTypeValue,
+		},
+	}, {
+		name: "union field",
+		inSrc: &copyTest{
+			UnionField: &copyUnionS{"new-belgium-fat-tire"},
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			UnionField: &copyUnionS{"new-belgium-fat-tire"},
+		},
+	}, {
+		name: "string slice",
+		inSrc: &copyTest{
+			StringSlice: []string{"sierra-nevada-pale-ale", "stone-ipa"},
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			StringSlice: []string{"sierra-nevada-pale-ale", "stone-ipa"},
+		},
+	}, {
+		name: "unimplemented string slice with existing members",
+		inSrc: &copyTest{
+			StringSlice: []string{"stone-and-wood-pacific", "pirate-life-brewing-iipa"},
+		},
+		inDst: &copyTest{
+			StringSlice: []string{"feral-brewing-co-hop-hog", "balter-brewing-xpa"},
+		},
+		wantDst: &copyTest{
+			StringSlice: []string{"stone-and-wood-pacific", "pirate-life-brewing-iipa", "feral-brewing-co-hop-hog", "balter-brewing-xpa"},
+		},
+		wantErr: true, // unimplemented.
```

Comment:
expand comment to "input combination not yet supported: dest slice must be nil at this time."
Also suggest removing wantDst unless you're planning to implement this feature soonish.
Here and below.

Reply:
Done.